src/mpi/romio/adio/common/ad_read_coll.c

```diff
@@ -774,7 +774,7 @@ static void ADIOI_R_Exchange_data(ADIO_File fd, void *buf, ADIOI_Flatlist_node
                                   ADIOI_Access * others_req, int iter, MPI_Aint buftype_extent,
                                   MPI_Aint * buf_idx, MPI_Aint * actual_recved_bytes)
 {
-    int i, nprocs_recv, nprocs_send;
+    int i, j, nprocs_recv, nprocs_send;
```

Comment:
I am not blocking this PR. But I would prefer to declare the scoped variables where it is used and move it inside the scope as much as we can. For example, `for`-loop allows us to write like this:
```
for (int i = 0, j = 0; i < nprocs; i++) {
    ...
    j++;
}
```